spec/lib/services/workflow-api-service-spec.js

```diff
@@ -376,6 +376,20 @@ describe('Http.Services.Api.Workflows', function () {
         });
     });
 
+    it('should return a sanitized workflow document', function(){
+        var workflow = {
+            id       : 'testgraphid',
+            _status  : 'pending',
+            password : 'testPassword'
```

Comment:
so I did fully vet the code, and it will traverse nested objects and arrays... I agree the unit tests can and will be expanded... that was just a place holder with a simple test.